src/kvstore/PartManager.cpp

```diff
@@ -188,5 +200,49 @@ void MetaServerBasedPartManager::fetchLeaderInfo(
     }
 }
 
+meta::ListenersMap MetaServerBasedPartManager::listeners(const HostAddr& host) {
+    auto ret = client_->getListenersByHostFromCache(host);
+    if (ret.ok()) {
+        return ret.value();
+    }
+    return meta::ListenersMap();
+}
+
+StatusOr<std::vector<meta::RemoteListenerInfo>>
+MetaServerBasedPartManager::listenerPeerExist(GraphSpaceID spaceId, PartitionID partId) {
+    return client_->getListenerHostTypeBySpacePartType(spaceId, partId);
+}
+
+void MetaServerBasedPartManager::onListenerAdded(GraphSpaceID spaceId,
+                                                 PartitionID partId,
+                                                 const meta::ListenerHosts& listenerHost) {
+    if (handler_ != nullptr) {
+        handler_->addListener(spaceId, partId, listenerHost.type_, listenerHost.peers_);
+    } else {
+        VLOG(1) << "handler_ is nullptr!";
```

Comment:
it's not an exception ?

Reply:
This path will only be hit in ut, same with other methods.